cuda/test/solver/upper_trs_kernels.cpp

```diff
@@ -174,21 +174,31 @@ TEST_F(UpperTrs, CudaMultipleRhsApplySyncfreeIsEquivalentToRef)
 TEST_F(UpperTrs, CudaMultipleRhsApplyIsEquivalentToRef)
 {
     initialize_data(50, 3);
+#if CUDA_VERSION >= 11031
+#if CUDA_VERSION < 12000
+    // The cuSPARSE Generic SpSM implementation uses the wrong stride here
+    // so the input and output stride need to match
+    auto in_stride = 4;
+    auto out_stride = 4;
+#else
+    // The cuSPARSE 12 solver is even worse: It only works if the stride is
+    // equal to the number of columns.
+    auto in_stride = 3;
+    auto out_stride = 3;
+#endif
+#else
+    auto in_stride = 4;
+    auto out_stride = 5;
+#endif
```

Comment:
This is a problem where the cusparse spsm is called, so maybe we should be handling it there instead?

Reply:
you mean throw an exception?

Reply:
Yes. I think a thrown exception with some nice message is better than some weird unreadable error, which cusparse exits with

Reply:
We don't throw an error right now, the results are just incorrect.

Reply:
Maybe we should throw an error ?